src/fliens/autocraft/AutoCraft.java

```diff
@@ -142,8 +150,8 @@ private boolean handleAutoCrafter(Block autocrafter) {
 				}
 			}
 			if (!destHasSpace)// success
-				return false;
-			destinationInv.addItem(result);
+				return;
+			destinationInv.addItem(result.clone()); // Fix for "Paper" implementation of addItem which rewrites the 'result'

```

Comment:
Does this also work with the other versions so spigot etc.?

Reply:
I'm pretty sure it does cuz this line of code is supposed to just grab the crafting result and add it to the output inventory, but as noted in bukkit reference, addItem is changing the given variable in some implementations which also changes the value in the cache. And this is just making the code use a copy of the variable to prevent unnecessary changes. I will test it on Spigot tomorrow and tell the results

Reply:
Ok so I tested my version on latest 1.16.5 Spigot and it works just fine. Had some troubles at first because I was compiling it with Java 16 which is supported by Paper 1.16.5 but not supported by Spigot 1.16.5. I recompiled it with Java 8 with no changes to the code and it...just worked. Also I tested your version from Spigotmc and found out that the same bug with overwriting cached result is also in the Spigot. First screenshot shows my version + server version and other stuff, second and third - wrong number of items being produced by the original plugin (it makes 8 iron instead of 9 from an iron block after the first stack is being filled up)
![2021-07-05_16 59 12](https://user-images.githubusercontent.com/34776474/124483663-44e9f580-ddb3-11eb-84e4-cc1e347fba71.png)
![2021-07-05_17 01 06](https://user-images.githubusercontent.com/34776474/124483772-61862d80-ddb3-11eb-9ef8-03ab38d76229.png)
![2021-07-05_17 01 07](https://user-images.githubusercontent.com/34776474/124483784-6519b480-ddb3-11eb-8f6a-a579d5d3d40b.png)

Reply:
Ok perfect

Reply:
Btw I can provide compiled jar so you can create a release

Reply:
I will do that myself tomorrow :)